### ISSUE TITLE
Atlas and Layout Creation improvements

### ISSUE THREAD

**Issue by AdmiralKew** [qgis/QGIS#33187]
As someone who uses the Atlas creator quite frequently, it's sometimes a bit of a pain to get everything to fit right when dealing with atlas features of varying shape and size. Sometimes one layout can work with one feature, but the next renders it invalid. I suggest:

- 'rotate to fit' in the 'controlled by atlas' rolldown in map properties. What this would do is rotate the map automatically to maximize the space used by the current feature.
- Alongside that, for full-page maps, could be a way to link titles, shapes, and legends to the map and have the layout window automatically move them so that the atlas feature is not covered up by these items, or if there is no other option, to minimize the area covered up by the item.

Another feature I'd like to add is the option to quickly set margins as guides.
- There could be a button to automatically place margins a user-inputted distance from the page edges,
- Simply inputting a negative value will start it from the opposite side of the page. For example, inputting 1" as a guide would start it on the left or top side. But -1" would place the guide on the right or bottom side of the page.

**Comment by roya0045** [comment]
You are aware that you can use the attributes of the feature to control such element as the rotation?

**Comment by AdmiralKew** [comment]
> You are aware that you can use the attributes of the feature to control such element as the rotation?

Yes, but such a thing is finicky and time-consuming when you're working with hundreds of individual maps to generate. Sorry for the late reply.

**Comment by Bernd-Vogelgesang** [comment]
>  'rotate to fit' in the 'controlled by atlas' rolldown in map properties. What this would do is rotate the map automatically to maximize the space used by the current feature.

Indeed, an automatic function for this will be great. So far I help myself with a model to prepare the data: I use the "minimum oriented bounding box" function, which generates kind of a coverage layer along with an "angle" attribute. This angle can be used to rotate the object so it best fits into your map frame.
In the "Map Rotation" of your Atlas, enter 
```
CASE
WHEN (90-"angle")> 0 THEN (90-"angle")
ELSE (180-"angle")
END
```
This will orient your north most of the time to the right direction and will not flip your objects upside down for landscape oriented pages.

Without this extra layer, one could maybe construct a similar rule based on this expression and other functions, which could do it on the fly:
For a start:
`angle_at_vertex(shortest_line(centroid(oriented_bbox($geometry)), boundary(oriented_bbox($geometry))),0)`
(found here: https://gis.stackexchange.com/a/352881)
